internal/resources/metal/project/datasource.go

```diff
@@ -121,12 +121,12 @@ func dataSourceMetalProjectRead(ctx context.Context, d *schema.ResourceData, met
 	if nameOK {
 		name := nameRaw.(string)
 
-		os, _, err := client.ProjectsApi.FindProjects(ctx).Execute()
+		projects, err := client.ProjectsApi.FindProjects(ctx).Name(name).ExecuteWithPagination()
```

Comment:
This is the line change that fixes #549.  Adopting `ExecuteWithPagination` on its own is enough to fix the bug: that function will load all projects available to the current token, as was done in the `packngo` implementation.  I added the `Name` filter as an optimization but can remove it if reviewers aren't comfortable with taking that risk.